components/movementsensor/rtkutils/cachedData.go

```diff
@@ -0,0 +1,176 @@
+package rtkutils
+
+import (
+	"context"
+	"math"
+	"sync"
+
+	"github.com/golang/geo/r3"
+	geo "github.com/kellydunn/golang-geo"
+	"github.com/pkg/errors"
+
+	"go.viam.com/rdk/components/movementsensor"
+	"go.viam.com/rdk/spatialmath"
+)
+
+var errNilLocation = errors.New("nil gps location, check nmea message parsing")
+
+// CachedGpsData allows the use of any MovementSensor chip via a DataReader.
+type CachedGpsData struct {
+	mu           sync.RWMutex
+	uncachedData GPSData
```

Comment:
not a big fan of the name "uncachedData". I think it should be named something like "gpsNmeaData"